mergify_engine/tests/unit/test_command.py

```diff
@@ -84,3 +92,178 @@ def test_command_loader_wuth_defaults():
+                {
+                    "id": 1,
+                    "name": "Justice League",
+                    "slug": "justice-league",
+                    "description": "A great team.",
+                    "privacy": "closed",
+                    "permission": "admin",
+                }
+            ],
+        },
+    )
+
+    rules = {}
+
+    with pytest.raises(RuntimeError) as error_msg:
+        await handle(
+            ctxt=ctxt, mergify_config=rules, comment="@Mergifyio update", user=None
+        )
+    assert "user must be set if rerun is false" in str(error_msg.value)
+
+
+class MockResponse:
+    @staticmethod
+    async def post():
+        return None
+
+
+@pytest.mark.asyncio
+async def test_run_command_with_bot(redis_cache, monkeypatch):
+
+    client = github.aget_client(owner_name="Mergifyio", owner_id=123)
+
+    sub = subscription.Subscription(
+        redis_cache,
+        123,
+        True,
+        "",
+        {},
+        frozenset({}),
+    )
+
+    installation = context.Installation(123, "Mergifyio", sub, client, redis_cache)
+
+    repository = context.Repository(installation, "demo", 123)
+
+    ctxt = await context.Context.create(
+        repository,
+        {
+            "number": 789,
+            "state": "open",
+            "title": "Amazing new feature",
+            "user": {
+                "login": "octocat",
+                "id": 1,
+            },
+            "mergeable_state": "ok",
+            "merged_by": None,
+            "merged": None,
+            "merged_at": None,
+            "base": {
+                "sha": "sha",
+                "ref": "main",
+                "user": {
+                    "login": {
+                        "Mergifyio",
+                    },
+                },
+                "repo": {
+                    "name": "demo",
+                    "private": False,
+                    "owner": {
+                        "login": "Mergifyio",
+                        "id": 123,
+                    },
+                    "permissions": {
+                        "admin": False,
+                        "push": False,
+                        "pull": True,
+                    },
+                },
+            },
+            "requested_teams": [
+                {
+                    "id": 1,
+                    "name": "Justice League",
+                    "slug": "justice-league",
+                    "description": "A great team.",
+                    "privacy": "closed",
+                    "permission": "admin",
+                }
+            ],
+        },
+        [],
+    )
+
+    user = github_types.GitHubAccount(
+        {
+            "id": config.BOT_USER_ID,
+            "login": "wall-e",
+            "type": "Bot",
+            "avatar_url": "https://avatars.githubusercontent.com/u/583231?v=4",
+        },
+    )
+
+    http_calls = []
+
+    async def mock_post(*args, **kwargs):
+        http_calls.append("http_call")
```

Comment:
You should put *args and **kwargs in the list of calls, and at the bottom assert on all arguments passed, to be sure this is the call you really want catch.